src/analysis/special_functions/exp_log.lean

```diff
@@ -389,107 +449,142 @@ end
+lemma strict_mono_decr_on_log : strict_mono_decr_on log (set.Iio 0) :=
 begin
-  rw tendsto_nhds_nhds, assume ε ε0,
-  let δ := min (exp ε - 1) (1 - exp (-ε)),
-  have : 0 < δ,
-    refine lt_min (sub_pos_of_lt (by rwa one_lt_exp_iff)) (sub_pos_of_lt _),
-      by { rw exp_lt_one_iff, linarith },
-  use [δ, this], assume x h,
-  cases le_total 1 x with hx hx,
-  { have h : x < exp ε,
-      rw [dist_eq, abs_of_nonneg (sub_nonneg_of_le hx)] at h,
-      linarith [(min_le_left _ _ : δ ≤ exp ε - 1)],
-    calc abs (log x - 0) = abs (log x) : by simp
-      ... = log x : abs_of_nonneg $ log_nonneg hx
-      ... < ε : by { rwa [← exp_lt_exp, exp_log], linarith }},
-  { have h : exp (-ε) < x,
-      rw [dist_eq, abs_of_nonpos (sub_nonpos_of_le hx)] at h,
-      linarith [(min_le_right _ _ : δ ≤ 1 - exp (-ε))],
-    have : 0 < x := lt_trans (exp_pos _) h,
-    calc abs (log x - 0) = abs (log x) : by simp
-      ... = -log x : abs_of_nonpos $ log_nonpos (le_of_lt this) hx
-      ... < ε : by { rw [neg_lt, ← exp_lt_exp, exp_log], assumption' } }
+  rintros x (hx : x < 0) y (hy : y < 0) hxy,
+  rw [← log_abs y, ← log_abs x],
+  refine log_lt_log (abs_pos.2 hy.ne) _,
+  rwa [abs_of_neg hy, abs_of_neg hx, neg_lt_neg_iff]
 end
 
-lemma continuous_log' : continuous (λx : {x:ℝ // 0 < x}, log x) :=
-continuous_iff_continuous_at.2 $ λ x,
+/-- The real logarithm function tends to `+∞` at `+∞`. -/
+lemma tendsto_log_at_top : tendsto log at_top at_top :=
+tendsto_comp_exp_at_top.1 $ by simpa only [log_exp] using tendsto_id
+
+lemma tendsto_log_nhds_within_zero : tendsto log (𝓝[{0}ᶜ] 0) at_bot :=
 begin
-  rw continuous_at,
-  let f₁ := λ h:{h:ℝ // 0 < h}, log (x.1 * h.1),
-  let f₂ := λ y:{y:ℝ // 0 < y}, subtype.mk (x.1 ⁻¹ * y.1) (mul_pos (inv_pos.2 x.2) y.2),
-  have H1 : tendsto f₁ (𝓝 ⟨1, zero_lt_one⟩) (𝓝 (log (x.1*1))),
-    have : f₁ = λ h:{h:ℝ // 0 < h}, log x.1 + log h.1,
-      ext h, rw ← log_mul (ne_of_gt x.2) (ne_of_gt h.2),
-    simp only [this, log_mul (ne_of_gt x.2) one_ne_zero, log_one],
-    exact tendsto_const_nhds.add (tendsto.comp tendsto_log_one_zero continuous_at_subtype_coe),
-  have H2 : tendsto f₂ (𝓝 x) (𝓝 ⟨x.1⁻¹ * x.1, mul_pos (inv_pos.2 x.2) x.2⟩),
-    rw tendsto_subtype_rng, exact tendsto_const_nhds.mul continuous_at_subtype_coe,
-  suffices h : tendsto (f₁ ∘ f₂) (𝓝 x) (𝓝 (log x.1)),
-  begin
-    convert h, ext y,
-    have : x.val * (x.val⁻¹ * y.val) = y.val,
-      rw [← mul_assoc, mul_inv_cancel (ne_of_gt x.2), one_mul],
-    show log (y.val) = log (x.val * (x.val⁻¹ * y.val)), rw this
-  end,
-  exact tendsto.comp (by rwa mul_one at H1)
-    (by { simp only [inv_mul_cancel (ne_of_gt x.2)] at H2, assumption })
+  have : tendsto abs (𝓝[{0}ᶜ] (0:ℝ)) (𝓝[set.Ioi 0] (abs 0)) :=
+    (continuous_abs.tendsto 0).inf (tendsto_principal_principal.2 $ λ a, abs_pos.2),
+  rw [abs_zero] at this,
+  simpa [(∘)] using (tendsto_neg_at_top_at_bot.comp $ tendsto_log_at_top.comp
+    tendsto_inv_zero_at_top).comp this
```

Comment:
```suggestion
  have h : tendsto log (𝓝[set.Ioi 0] 0) at_bot :=
    tendsto_comp_exp_at_bot.1 (by simpa only [log_exp] using tendsto_id),
  simpa [(∘)] using h.comp this,
```
I think you don't need this particular trick (inverting and re-inverting).  My suggestion is still longer than yours with the trick, but maybe you can golf it ...